apps/src/templates/progress/ProgressLesson.jsx

```diff
@@ -211,28 +215,29 @@ class ProgressLesson extends React.Component {
               {hiddenForStudents && (
                 <FontAwesome icon="eye-slash" style={styles.icon} />
               )}
-              {showLockIcon && lesson.lockable && locked && (
-                <FontAwesome icon="lock" style={styles.icon} />
-              )}
-              {showLockIcon && lesson.lockable && !locked && (
-                <span data-tip data-for={tooltipId}>
+              {lesson.lockable && (
+                <span data-tip data-for={lockedTooltipId}>
                   <FontAwesome
-                    icon="unlock"
+                    icon={
+                      isLockedForSection || isLockedForUser ? 'lock' : 'unlock'
```

Comment:
(minor) I think it might be worth having a local variable (specifically for display purposes) whose value is `isLockedForSection || isLockedForUser`.  (It's great that we keep these two cases separate in most of the stack to clarify that these are separate cases; the variable would make it clear that we choose to _display_ these two cases in the same way to the student/teacher.)